src/armor_layers.cpp

```diff
@@ -112,387 +106,499 @@ item_penalties get_item_penalties( std::list<item>::const_iterator worn_item_it,
+            return itm_iter == o.itm_iter;
+        }
 
-struct layering_item_info {
-    item_penalties penalties;
-    int encumber;
-    std::string name;
-
-    // Operator overload required to leverage vector equality operator.
-    bool operator ==( const layering_item_info &o ) const {
-        // This is used to merge e.g. both arms into one entry when their items
-        // are equivalent.  For that purpose we don't care about the exact
-        // penalities because they will list different body parts; we just
-        // check that the badness is the same (which is all that matters for
-        // rendering the right-hand list).
-        return this->penalties.badness() == o.penalties.badness() &&
-               this->encumber == o.encumber &&
-               this->name == o.name;
-    }
+        int get_col_value( sort_armor_columns col ) {
+            item itm = *itm_iter;
+            int ret_val;
+            switch( col ) {
+                case column_warmth:
+                    ret_val = itm.get_warmth();
+                    break;
+                case column_coverage:
+                    ret_val = itm.get_coverage();
+                    break;
+                case column_bash:
+                    ret_val = itm.bash_resist();
+                    break;
+                case column_cut:
+                    ret_val = itm.cut_resist();
+                    break;
+                case column_acid:
+                    ret_val = itm.acid_resist();
+                    break;
+                case column_fire:
+                    ret_val = itm.fire_resist();
+                    break;
+                case column_env:
+                    ret_val = itm.get_env_resist();
+                    break;
+                default:
+                    ret_val = 0;
+                    break;
+            }
+            return ret_val;
+        }
+        void print_columnns( bool is_selected, const catacurses::window &w, int right_bound,
+                             int cur_print_y ) {
+            item itm = *itm_iter;
+            nc_color print_col = is_selected ? hilite( c_white ) : c_light_gray;
+            for( size_t col_idx = column_num_entries; col_idx-- > 0; ) {
+                int col_w = get_col_data( col_idx ).get_width();
+                right_bound -= col_w;
+                switch( col_idx ) {
+                    case column_cond: {
+                        nc_color print_col2 = is_selected ? hilite( c_red ) : c_red;
+                        if( is_selected ) { //fill whole line with color
+                            mvwprintz( w, point( right_bound, cur_print_y ), print_col2, "%*s", col_w, "" );
+                        }
+                        int pos_x = right_bound + col_w - 1;
+                        char cond;
+
+                        print_col2 = is_selected ? hilite( c_cyan ) : c_cyan;
+                        cond = ( itm.is_filthy() ? 'f' : ' ' );
```

Comment:
```suggestion
                        const char cond = ( itm.is_filthy() ? 'f' : ' ' );
```
And remove the definition of `cond` above.